src/psyclone/tests/lfric_build_test.py

```diff
@@ -0,0 +1,152 @@
+import subprocess
+import pytest
+
+from psyclone.tests.lfric_build import LFRicBuild
+from psyclone.tests.utilities import Compile, CompileError
+
+
+@pytest.fixture(name="enable_compilation")
+def protect_infrastructure_path_fixture(monkeypatch):
+    '''A pytest fixture that mocks testing with compilation enabled while
+    ensuring that any pre-existing LFRicBuild._compilation_path is
+    preserved.'''
+    # Pretend that the infrastructure has already been built so that the
+    # LFRicBuild constructor doesn't attempt to trigger it.
+    monkeypatch.setattr(LFRicBuild, "_infrastructure_built", True)
+    # Pretend that compilation testing is enabled.
+    if not Compile.TEST_COMPILE:
+        monkeypatch.setattr(Compile, "TEST_COMPILE", True)
+    else:
+        # If compilation testing is enabled then the infrastructure lib will
+        # already have been built so we use monkeypatch to ensure its
+        # location is automatically restored at the end of this test.
+        monkeypatch.setattr(LFRicBuild, "_compilation_path", "/no/path")
+
+
+def test_lf_build_get_infrastructure_flags(monkeypatch, tmpdir):
+    '''
+    Test the get_infrastructure_flags method.
+
+    '''
+    # Pretend that compilation testing is disabled.
+    monkeypatch.setattr(Compile, "TEST_COMPILE", False)
+    builder = LFRicBuild(tmpdir)
+    flags = builder.get_infrastructure_flags()
+    dir_list = []
+    for idx, flag in enumerate(flags):
+        if idx % 2 == 0:
+            assert flag == '-I'
+        else:
+            # Just keep a list of the base directories
+            dir_list.append(os.path.split(flag)[1])
+    assert 'configuration' in dir_list
+    assert 'function_space' in dir_list
+    assert 'field' in dir_list
+
+
+@pytest.mark.usefixtures("enable_compilation")
+def test_lfric_build_compiler_flags(tmpdir, monkeypatch):
+    '''
+    Check that the compiler settings supplied to pytest are passed through
+    when building the (stub) LFRic infrastructure.
+
+    '''
+    def fake_popen(arg_list, stdout=None, stderr=None):
+        '''Mock implementation of Popen that just raises an Exception.'''
+        raise OSError(f"arg_list = {arg_list}")
+
+    # Monkeypatch subprocess.Popen so that it just raises an exception.
+    monkeypatch.setattr(subprocess, "Popen", fake_popen)
+    builder = LFRicBuild(tmpdir)
+    # Set-up a custom compiler and flags.
+    builder._f90 = "my_compiler"
+    builder._f90flags = "-my -special -flags"
+    # Finally, check that these get passed through to the call to Popen when
+    # the infrastructure is built.
+    with pytest.raises(CompileError) as err:
+        builder._build_infrastructure()
+    assert ("['make', 'F90=my_compiler', 'F90FLAGS=-my -special -flags', '-f'"
+            in str(err.value))
+
+
+@pytest.mark.usefixtures("enable_compilation")
+def test_lfric_build_infrastructure(tmpdir, monkeypatch):
+    '''
+    Test the _build_infrastructure method when compilation appears to proceed.
```

Comment:
Can you explain a bit better what exactly this tests? It doesn't really test the infrastructure build, I think(?) it tests that error handling in the infrastructure works?? I.e. errors are detected, and non-errors won't raise anything?

Reply:
I've had a go - let me know if you think it needs more.

Reply:
That's great, thanks.